src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private int currentCapacity;
 
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        currentCapacity = INITIAL_CAPACITY;
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        size = 0;
+    }
     @Override
     public void put(K key, V value) {
+        int index = calculateIndex(key);
+        Node<K,V> newNode = new Node<>(key, value, null);
+        table[index] = putANewNode(index, table, newNode);
+    }
 
+    private int calculateIndex(K key) {
+        int index = (key != null)? (key.hashCode() % currentCapacity) : 0;
```

Comment:
The calculation of the index using `key.hashCode() % currentCapacity` can result in a negative index if the hash code is negative. Although this is handled by taking the absolute value, it might be more efficient to use `Math.floorMod(key.hashCode(), currentCapacity)` to directly compute a non-negative index.